src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -2996,25 +3001,103 @@ def get_enable_secret_rotation(self) -> bool:
 
         # read the original value passed by the command
         enable_secret_rotation = self.raw_param.get("enable_secret_rotation")
-        # try to read the property value corresponding to the parameter from the `mc` object
-        if (
-            self.mc and
-            self.mc.addon_profiles and
-            CONST_AZURE_KEYVAULT_SECRETS_PROVIDER_ADDON_NAME in self.mc.addon_profiles and
-            self.mc.addon_profiles.get(
-                CONST_AZURE_KEYVAULT_SECRETS_PROVIDER_ADDON_NAME
-            ).config.get(CONST_SECRET_ROTATION_ENABLED) is not None
-        ):
-            enable_secret_rotation = self.mc.addon_profiles.get(
-                CONST_AZURE_KEYVAULT_SECRETS_PROVIDER_ADDON_NAME
-            ).config.get(CONST_SECRET_ROTATION_ENABLED) == "true"
+        # In create mode, try to read the property value corresponding to the parameter from the `mc` object.
```

Comment:
Currently we are refactoring `aks create` and `aks update`, and "In create mode" means this only applies to the case of `aks create` but not `aks update`.